### PR TITLE
Add support for autoformatting comprehensions

### PR DESCRIPTION
**Pull request by nielsdebruin**
## What's changed?

The AutoFormatter is now able to format list, dict, and set comprehensions

## What's your motivation?

Automagically, have nice code!

## Anyone you would like to review specifically?

@knutwannheden 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: rewrite/rewrite/python/format/spaces_visitor.py [hunk 1]
```diff
@@ -408,6 +408,38 @@ def visit_expression_type_tree(self, expression_type_tree: ExpressionTypeTree, p
         ett = space_before(ett, False)
         return ett
 
+    def visit_comprehension_expression(self, comprehension_expression: ComprehensionExpression, p: P) -> J:
+        ce = cast(ComprehensionExpression, super().visit_comprehension_expression(comprehension_expression, p))
+
+        # Handle space before result this will depend on the style setting for the comprehension type.
+        if ce.kind in (ComprehensionExpression.Kind.LIST, ComprehensionExpression.Kind.GENERATOR):
```
Comment (knutwannheden) [inline rewrite/rewrite/python/format/spaces_visitor.py:415]: For generator expressions like `gen_exp = (x ** 2 for x in range(5))` I get the impression that none of the spaces styles actually applies. Instead PyCharm will just remove the spaces after the opening brace and before the closing brace (at least if it doesn't contain new lines or comments).

Comment (nielsdebruin) [reply rewrite/rewrite/python/format/spaces_visitor.py]: I have modified them also to do the same.

File: rewrite/rewrite/python/format/spaces_visitor.py [hunk 2]
```diff
@@ -408,6 +408,41 @@ def visit_expression_type_tree(self, expression_type_tree: ExpressionTypeTree, p
...
+    def visit_comprehension_clause(self, clause: ComprehensionExpression.Clause, p: P) -> J:
+        cc = cast(ComprehensionExpression.Clause, super().visit_comprehension_clause(clause, p))
+
+        # Ensure single space before 'for' keyword
+        cc = space_before(cc, True)
+
+        # Space around iterator variable and after 'in' keyword e.g. for  i  in  range(10) <-> ...for i in range(10)
+        cc = cc.padding.with_iterated_list(space_before_left_padded(cc.padding.iterated_list, True))
+        cc = cc.with_iterator_variable(space_before(cc.iterator_variable, True))
+        cc = cc.padding.with_iterated_list(space_before_left_padded_element(cc.padding.iterated_list, True))
```
Comment (knutwannheden) [inline rewrite/rewrite/python/format/spaces_visitor.py:443]: Can you verify that the first and last of these statements are both required? They look very similar and I don't know what the difference is, really.

Comment (nielsdebruin) [reply rewrite/rewrite/python/format/spaces_visitor.py]: Yes, I have added some extra comments to their exact purpose. For extra validation, you can also comment on one, which will result in the tests failing.

**Pull request merged**